### ISSUE TITLE
ImGUI filebrowser: confusing: icon to clear the filter is identical to the icon to go to the current directory

### ISSUE THREAD

**Issue by MBilderbeek** [openMSX/openMSX#1651]
See upstream issue: https://github.com/aiekick/ImGuiFileDialog/issues/182

Feedback from upstream is: just use a different icon, you have to specify the icons yourself.

How to info:
- icons in ImGUI are glyphs in a font
- https://github.com/aiekick/ImGuiFontStudio can be used to create/edit such fonts
-  src/3rdparty/ImGuiFileDialog/CustomFont.{h,ii,ifs,png} have to be adjusted for this (only the first 2 are really used)
- Starting point could be font-Awesome (or Fork-Awesome). The current icons also originate from there.
- we assume anyone can use ImGUiFontSTudio to load a font (.ttf?), select a subset of the glyphs and then export to the 4 files we need

**Comment by MBilderbeek** [comment]
![image](https://github.com/openMSX/openMSX/assets/12675978/681d93c5-bc64-4188-bcca-b64ac2d10842)
This is a better alternative to replace the 'search' icon, as this is a filter, what it really is.

As how to clear the filter, not sure what the best icon is. Perhaps a combination of undo and filter? 
![image](https://github.com/openMSX/openMSX/assets/12675978/0737d3af-c263-44b9-9222-abea3ffab3b6)
Or just the undo icon on itself?

Anyway, I am not sure how to generate the .h and .ii files... it seems it can only generate cpp files but I can't even find these back after "Generate"....

**Comment by m9710797** [comment]
Maybe swap the order of the filter and the clear-filter icon, and then use a "X" (a big cross) for the latter? I mean not everything needs a dedicated icon, from the context and from the tool-tip it's also clear what a button does.

I just tried myself, the ImGuiFontStudio GUI is indeed confusing, but I did manage to export the required files:
* In the "Generator" pane, in the section "Global Settings (All Fonts)". For "Features" I enable "Header", "Card" and "Src". For "Settings" I select "C++".
* In the "Settings Per Font (Batch mode)" section, I make the same selection: "H" (for header), "C" (for card), "S" (for src) and "C++".  I have no idea what the difference between this and the previous stuff is, and if both are needed or not.
* After this there's now appears a "Generate" button in the "Generation" section.

If I remember correctly the ".ii" file is just a rename (+ small edit?) of the ".cpp" file.

--

While you're working on this. It would be nice if you could immediately include the icons for the debugger toolbar in this new custom font (then we only need to load one extra custom font).

**Comment by MBilderbeek** [comment]
I've updated the icons a bit. But instead of unduplicating the icons for these both 'reset' buttons, I changed the icon to a more intuitive 'undo' icon and for the rest, I think the user should simply rely on the tool tip to see the difference. In hindsight, I don't think this is a problem. 

Thanks @m9710797 for your assistance. I got things going (with some manual tweaking... somehow the program generated u8 prefixes in CustomFont.h, which I had to manually remove to get things working).
